### PR TITLE
fix: Making R optional

### PR DESCRIPTION
**Pull request by Kolarovszki**
So far, one could not run the tests without installing R, because `rpy2`
couldn't be installed without R. By creating a separate
`optional-requirements.txt` file with a single entry `rpy2`, the
optional development dependencies are separated.

The `ignore` pytest marker is deleted, since one can use builtin
`skipif` marker. Using this, the conditional execution of
`test_haar_random_interferometers_distance_for_ccr_strategy` is written.

The `rpy2` dependency handling of `cliffords_r_simulation_strategy` is
deleted entirely, and moved to `simulation_strategy_factory`, since it
requires less boilerplate code doing this way.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -17,13 +17,12 @@ python =
 [testenv]
 whitelist_externals = git
 # In case requirements.txt got updated, one wants to recreate (it was not automatical)
-recreate = True 
+recreate = True
 deps = -rrequirements.txt
 commands =
 	git submodule sync -q
```
Comment (Tomev) [inline tox.ini:23]: There are no submodules in theboss now, so git commands can also be removed. Would you mind doing it?

File: theboss/simulation_strategies/simulation_strategy_factory.py [hunk 2]
```diff
@@ -4,10 +4,10 @@
     This file holds a factory for simulation strategies.
 """
 
+from ast import Import
```
Comment (Tomev) [inline theboss/simulation_strategies/simulation_strategy_factory.py:7]: This import is unused.

File: theboss/simulation_strategies/simulation_strategy_factory.py [hunk 3]
```diff
@@ -138,12 +138,35 @@ def _generate_uniform_losses_strategy(self) -> UniformLossSimulationStrategy:
             self._experiment_configuration.uniform_transmissivity
         )
 
-    def _generate_r_cliffords_strategy(self) -> CliffordsRSimulationStrategy:
```
Comment (Tomev) [inline theboss/simulation_strategies/simulation_strategy_factory.py:141]: Why did you remove the return type? It's still the same (but the method my raise some errors). I think it should stay.

**Review by Tomev — APPROVED** [review]
I've got a few nitpicks, otherwise it looks good.

File: theboss/simulation_strategies/simulation_strategy_factory.py (hunk 3 above)
Comment (Kolarovszki) [reply theboss/simulation_strategies/simulation_strategy_factory.py:141]: I couldn't have it here, since the class `CliffordsRSimulationStrategy` is not imported at this stage yet.

File: theboss/simulation_strategies/simulation_strategy_factory.py (hunk 2 above)
Comment (Kolarovszki) [reply theboss/simulation_strategies/simulation_strategy_factory.py]: done

File: tox.ini (hunk 1 above)
Comment (Kolarovszki) [reply tox.ini]: done
